### ISSUE TITLE
[18.1.1][BUG][`clang-tidy`][`bugprone-inc-dec-in-conditions`] `decltype` should not count as a variable "reference"

### ISSUE THREAD

**Issue by Jacobfaib** [llvm/llvm-project#85838]
The new `bugprone-inc-dec-in-conditions` should not consider `decltype()` to be a reference of a variable when emitting this check.
```c++
int main()
{
    auto foo = 0;
    auto bar = 0;
    if (++foo < static_cast<decltype(foo)>(bar)) { }
    return 0;
}
```
```
[<source>:6:9: warning: incrementing and referencing a variable in a complex condition can cause unintended side-effects due to C++'s order of evaluation, consider moving the modification outside of the condition to avoid misunderstandings [bugprone-inc-dec-in-conditions]](javascript:;)
    6 |     if (++foo < static_cast<decltype(foo)>(bar)) { }
      |         ^
[<source>:6:38: note: variable is referenced here](javascript:;)
    6 |     if (++foo < static_cast<decltype(foo)>(bar)) { }
      |                                      ^
1 warning generated.
```
https://godbolt.org/z/6ahz8qea7

Expected Behavior
-----------------
No diagnostics emitted for the above code, or, if this check intends to catch `++foo` itself, have the diagnostic reference the increment itself.

**Comment by PiotrZSL** [comment]
Will be fixed in 19.0.
